docs/src/query_interface.md

```diff
@@ -0,0 +1,120 @@
+# Query Interface
+
+```@meta
+DocTestSetup =  quote
+    using QuantumSavory
+end
+```
+
+
+## `Tag`
+Tags are identifiers which are used to represent classical information needed for a quantum information. The library allows the construction of custom tags following the format of one of the [`tag_types`](@ref) using the `Tag` constructor. The library implements the following tags for use in the networking protocols:
+
+- `EntanglementCounterpart` 
+    -`remote_node`
+    -`remote_slot`
+    It indicates the current entanglement status with a remote node's slot.
+
+- `EntanglementHistory`
+    - `remote_node`
+    - `remote_slot`
+    - `swap_remote_node`
+    - `swap_remote_slot`
+    - `swapped_local`
+    This tag is used to store the outdated entanglement information after a swap. It helps to direct incoming entanglement update messages to the right node after a swap.
+
+- `EntanglementUpdateX`
+    - `past_local_node`
+    - `past_local_slot`
+    - `past_remote_slot`
+    - `new_remote_node`
+    - `new_remote_slot`
+    - `correction`
+    This tag arrives as a message from a remote node to which the current node was entangled to updat the entanglement information and apply an `X` correction after the remote node performs an entanglement swap.
+
+- `EntanglementUpdateZ`
+    - `past_local_node`
+    - `past_local_slot`
+    - `past_remote_slot`
+    - `new_remote_node`
+    - `new_remote_slot`
+    - `correction`
+    This tag arrives as a message from a remote node to which the current node was entangled to updat the entanglement information and apply a `Z` correction after the remote node performs an entanglement swap.
+
+The tags are constructed using the `Tag` constructor
+#### Tag(tagsymbol::Symbol, tagvariants...)
+where `tagvariants` are the extra arguments required by the specific `tagsymbol`, for instance the `tag_types.SymbolIntInt` require two `Int` values. It supports the use of predicate functions (`Int -> Bool`) and [`Wildcard`](@ref) (❓) in place of the `tagvariants` which allows the user to perform queries for tags fulfilling certain criteria.
+
+## `tag!`
+Adds a `Tag` to the list of tags associated with a [`RegRef`](@ref) in a [`Register`](@ref)
+#### `tag!(ref::RegRef, tag::Tag)`
```

Comment:
These manual listings of method signatures should be automated otherwise they will become outdated very quickly. It is important to NOT rely on humans to keep documentation and code in sync -- that never works.